### PR TITLE
[WIP] metasploit-framework added

### PR DESCRIPTION
**Pull request by Auxilus**
This PR is for testing purpose for now..

### CODE REVIEW

**Comment by Auxilus** [comment]
As https://github.com/termux/termux-packages/pull/1361/commits/e4f11b7c47ed498c63f4133f3bc26513d5d751a8 is running without any errors maybe we have to work around with that format...

File: packages/metasploit-framework/build.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        gem build grpc.gemspec
+        gem install grpc-1.4.1.gem
+        cd ..
+        rm -r grpc-1.4.1
+
+
+        cd $TERMUX_PREFIX/share/metasploit-framework
+        bundle install -j5
+
+        ln -s $TERMUX_PREFIX/share/metasploit-framework/msfconsole /data/data/com.termux/files/usr/bin/
```
Comment (stephengroat) [inline packages/metasploit-framework/build.sh]: ln -s $TERMUX_PREFIX/share/metasploit-framework/msfconsole **$TERMUX_PREFIX/bin/**

**Comment by Grimler91** [comment]
I think you can still skip all the dependencies, at least for testing purposes.

The targeted platform of all the gem installations should be set to $TERMUX_ARCH or something similar so that it changes when we build for arm or aarch64 etc.
All of them should also install to $TERMUX_PREFIX/lib/ruby/gems/2.4.0 (I see that there is a forgotten "$" sign there).

**Comment by Auxilus** [comment]
how can I access metasploit files inside the build?
Is there something like TERMUX_PKG_DIR?

**Comment by Grimler91** [comment]
$TERMUX_PKG_SRCDIR is the path where the metasploit tar.gz file has been extracted. Which files are you referring to?

**Comment by Auxilus** [comment]
Aha, atleast cding and symlinking is working

**Comment by Auxilus** [comment]
I'd have to add dependencies for nokogiri, but i can try grpc without any dependencies

**Comment by Auxilus** [comment]
This commit ends up implementing all the patching and gem unpacking stuff..

**Comment by Auxilus** [comment]
https://github.com/termux/termux-packages/pull/1361/commits/c9e52729ac366130c05d4aa8a949cee72b11332d ends up implementing the while loop for gem installation (it fails cz we haven't yet added the dependencies to build.sh)

**Comment by Auxilus** [comment]
@fornwall I think PR is ready to get merged, you can try building the package, if you encounter any error please let me know.

**Comment by Auxilus** [comment]
Getting `Building native extensions.  This could take a while...                                                                                                                                                                                        
ERROR:  Error installing actionpack:                                                                                                                                                                                                           
        ERROR: Failed to build gem native extension.                                                                                                                                                                                           
                                                                                                                                                                                                                                               
    current directory: /data/data/com.termux/files/usr/lib/ruby/gems/2.4.0/gems/json-1.8.6/ext/json/ext/generator                                                                                                                              
/usr/bin/ruby2.3 -r ./siteconf20170908-25515-mik8bb.rb extconf.rb                                                                                                                                                                              
mkmf.rb can't find header files for ruby at /usr/lib/ruby/include/ruby.h                                                                                                                                                                       
                                                                                                                                                                                                                                               
extconf failed, exit code 1           ' while building, is it due to wrong dir for rubg.h? If yes then how do I change it to correct one?

**Comment by Auxilus** [comment]
I'm having some issue with the build regarding gem installation. If anyone has any experience with gem installation or how it works, your suggestions are appreciated.
Take a look at the [build logs](http://sprunge.us/GXjB) and the build script itself.

**Comment by zhengyo** [comment]
我装不了msf

**Pull request closed without merging**